### PR TITLE
extract broker in separate component

### PR DESCRIPTION
**Pull request by davidescus**

### CODE REVIEW

File: broker/cmd/dejaqcli-broker/main.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	"github.com/dejaq/prototype/grpc/DejaQ"
+	flatbuffers "github.com/google/flatbuffers/go"
+	_ "github.com/lib/pq"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+type Config struct {
+	Host         string `env:"HOST" env-default:"127.0.0.1:9000"`
+	OverseerSeed string `env:"OVERSEER" env-default:"localhost:9000"`
```
Comment (bgadrian) [inline broker/cmd/dejaqcli-broker/main.go:30]: the broker is the overseer

File: broker/cmd/dejaqcli-broker/main.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	storageTimeline "github.com/dejaq/prototype/broker/pkg/storage/timeline"
+	"github.com/dejaq/prototype/grpc/DejaQ"
+	flatbuffers "github.com/google/flatbuffers/go"
+	_ "github.com/lib/pq"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+type Config struct {
+	Host         string `env:"HOST" env-default:"127.0.0.1:9000"`
```
Comment (bgadrian) [inline broker/cmd/dejaqcli-broker/main.go:29]: This is usually called binding address, the host is the actually machine that is running on

File: broker/cmd/dejaqcli-broker/main.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	OverseerSeed string `env:"OVERSEER" env-default:"localhost:9000"`
+	StorageType  string `env:"STORAGE_TYPE" env-default:"memory"`
+	StorageHost  string `env:"STORAGE_HOST"`
+
+	TopicCount        int `env:"TOPIC_COUNT"`
```
Comment (bgadrian) [inline broker/cmd/dejaqcli-broker/main.go:34]: I suggest to remove these 3 settings and the concurrent Streams limitations, or set a specific config for it

File: broker/cmd/dejaqcli-broker/main.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	if c.TopicCount < 1 {
+		return errors.New("TopicCount should be > 0")
+	}
+	if c.ConsumersPerTopic < 1 {
+		return errors.New("ConsumersPerTopic should be > 0")
+	}
+	if c.ProducersPerTopic < 1 {
+		return errors.New("ProducersPerTopic should be > 0")
+	}
+	if _, err := time.ParseDuration(c.TimeoutDuration); err != nil {
```
Comment (bgadrian) [inline broker/cmd/dejaqcli-broker/main.go:62]: timeout should not be mandatory, only check if is not empty

File: broker/cmd/dejaqcli-broker/main.go (hunk 3 above)
Comment (davidescus) [reply broker/cmd/dejaqcli-broker/main.go:34]: Remove those settings, add "MaxConnectionsLimit"

**Pull request merged**